astroid/bases.py

```diff
@@ -536,7 +563,7 @@ def _infer_type_new_call(self, caller, context):  # noqa: C901
             mcs = next(caller.args[0].infer(context=context))
         except StopIteration as e:
             raise InferenceError(context=context) from e
-        if mcs.__class__.__name__ != "ClassDef":
+        if not isinstance(mcs, nodes.ClassDef):
```

Comment:
This is a behavior change for subclasses of nodes.ClassDef ?

Reply:
There's more such examples after that but I did not comment on each one (with Const for example). Maybe sometime it's important because msc could actually be a subclass ? Or maybe it's a fix because subclasses were supposed to be included in this conditional ?

Reply:
Yeah, I think as long as tests pass this should be fine.

It very much feels like this was a relic of the `lazy_import` stuff where `nodes` was previously unavailable.
Using `name` is bad anyway as we would match against any object that is also called ``Name``